### ISSUE TITLE
Selects in Safari are styled differently to other browsers

### ISSUE THREAD

**Issue by matteason** [alphagov/govuk-frontend#3520]
`<select>`s in Safari retain quite a lot of default styling and so appear inconsistent with other form components, and with how selects appear in other browsers.

iOS 16.2 (Safari and Chrome, because Chrome uses Webkit on iOS):

![A GOV.UK-style select in iOS Safari with rounded corners](https://user-images.githubusercontent.com/1935173/232555588-4c5b3e95-1862-4b57-a955-a01905b2fc39.png)

Safari 16.1 on macOS Ventura:

![A GOV.UK-style select in mac OS Safari with a glossy appearance](https://user-images.githubusercontent.com/1935173/232555905-f7b40148-c020-4560-8e40-563eb2966e35.png)

(Chrome desktop, for reference)

![A GOV.UK-style select in Chrome with square corners](https://user-images.githubusercontent.com/1935173/232560523-b69a9815-f8b1-442b-ad34-6c6091cd25bd.png)

The rounded corners on iOS are quite easy to overcome with `border-radius: 0`

![A GOV.UK-style select in iOS Safari with square corners](https://user-images.githubusercontent.com/1935173/232557315-f6022f8c-b971-47e2-b687-ee50d53809da.png)

Unfortunately macOS is trickier. `border-radius` has no effect unless you also add `-webkit-appearance: none` - but that also removes the arrows:

![A GOV.UK-style select in mac OS Safari with a white background and square corners, but no arrows indicating it can be opened](https://user-images.githubusercontent.com/1935173/232558311-f551e3cb-23a5-4ca1-8545-6fd7cd654ca3.png)

[Digital Scotland Design System's select component](https://designsystem.gov.scot/components/select/) handles this by introducing a wrapper element and using a styled span for the dropdown arrow:

![image](https://user-images.githubusercontent.com/1935173/232559148-e2f79cad-8293-4e3f-91f0-153e36ac42df.png)

The span is overlaid on the `<select>` with `position: absolute` and has `pointer-events: none` so clicks just fall through to the `<select>`

This approach isn't ideal and would be a breaking change for the GOV.UK Frontend (but I wonder if the same effect could be achieved with an `:after` instead?)

See also https://github.com/alphagov/govuk-frontend/issues/2435 for previous issues with selects retaining too much default styling in Firefox and Safari

**Comment by matteason** [comment]
Yeah it's definitely a tricky problem. I didn't know you can't use `::after` on selects, that does complicate it. Some people use [background images](https://stackoverflow.com/a/42322419) but that's going to cause accessibility issues in high contrast mode when the background disappears.

Obviously browser usage will vary by service, but on GOV.UK [Safari & Chrome on iOS are about 30% of sessions and 37% of users](https://webperf.social/@TheRealNooshu@hachyderm.io/109947374000950223). Safari on Mac is much lower (4% and 5%). Given that the solution for iOS is straightforward (`border-radius: 0`) perhaps that's the best that could be done for now, then subsequent improvements for Mac would probably depend on improved CSS features being available, or Apple changing the default styling. [Inputs also explicitly set border radius](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/input/_index.scss#L21).

**Comment by 36degrees** [comment]
There's a bit of related discussion in [the backlog issue for selects](https://github.com/alphagov/govuk-design-system-backlog/issues/60) – including links to previous attempts to style selects and the issues we ran into.
